### PR TITLE
Add jupyterlab message

### PR DESCRIPTION
**Pull request by ChaamC**
PR for [DAC-438](https://crim-ca.atlassian.net/browse/DAC-438) 

Changes:
--------
- Add a JupyterLab extension to display a message on the top bar, along with the file `plugin.json` to customize the 
  extension display.

Fixes:
------
- Fix bug while building JupyterLab (see [issue](https://github.com/jupyterlab/jupyterlab/issues/11248)).
- Fix notebook scripts to use a specific commit id.

Since, the solution that uses [JupyterHub templates](https://jupyterhub.readthedocs.io/en/stable/reference/templates.html) cannot add a message to the JupyterLab page but only works for actual JupyterHub pages,  this other solution was implemented. It uses an existing JupyterLab extension that can display a message on the top bar. 
Here is an overview of how it looks :

image

Note that there are some limitations. The display can be messed up if the window's width is too small. Some of the options at the top left can be hidden :

image

Also, right-clicking on the top bar displays the disabled option `Edit Text`, which let's the user change the displayed text and which was disabled explicitly via the `plugin.json` file, and an indication to the user on how to use the actual browser's right click menu instead when right clicking on the top bar :

image

If this solution does not satisfy our requirements, a simple `README` on the user's workspace could be another idea. Else, we would have to make our own JupyterLab extension which would require much more work.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,11 +6,14 @@ Unreleased (latest)
...
 Changes:
 --------
-- ...
+- Add a JupyterLab extension to display a message on the top bar, along with the file `plugin.json` to customize the 
+  extension display.
 
 Fixes:
 ------
-- ...
+- Fix bug while building JupyterLab (see [issue](`https://github.com/jupyterlab/jupyterlab/issues/11248`)).
```
Comment (fmigneault) [inline CHANGELOG.md]: No <code>`</code> around the link in Markdown.

File: schemas/plugin.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "editable": {
+      "title": "Editable",
+      "description": "Whether the text can be edited via the UI",
+      "default": false,
+      "type": "boolean"
+    }
+  },
+  "additionalProperties": false,
+  "type": "object"
+}
```
Comment (fmigneault) [inline schemas/plugin.json]: missing newline

File: Dockerfile [hunk 3]
```diff
@@ -47,11 +50,19 @@ RUN jupyter labextension install @jupyter-widgets/jupyterlab-manager \
...
-ADD https://raw.githubusercontent.com/jupyter/docker-stacks/master/base-notebook/fix-permissions /usr/local/bin/
-ADD https://raw.githubusercontent.com/jupyter/docker-stacks/master/base-notebook/jupyter_notebook_config.py /etc/jupyter/
+# Install extension to display custom message on JupyterLab's topbar
+RUN jupyter labextension install jupyterlab-topbar-extension \
+    && jupyter labextension install jupyterlab-topbar-text
+COPY schemas/plugin.json /opt/conda/envs/birdy/share/jupyter/lab/schemas/jupyterlab-topbar-text/plugin.json
+
+# Use an older fixed commit id, since the repo structure has changed, and using the newer scripts results in a bug with our current setup
+# Fixed to the commit id used for the base image 0.2.2 for now
+ENV DOCKER_STACKS_COMMIT_ID 2ddf41a430e15238afccbd4476fd8087d3252fb0
```
Comment (fmigneault) [inline Dockerfile:60]: Has the `0.2.2` image not been rebuilt for a while? 
This commit seems >1 year old (Sep 13, 2021).
https://github.com/jupyter/docker-stacks/commit/2ddf41a430e15238afccbd4476fd8087d3252fb0

**Comment by tlvu** [comment]
Oh interesting solution to a problem we also have.

When we release a new version of the Jupyter env, we email all the Jupyter users notifying them of the new release and include instructions on how to restart their Jupyter server to get the new version.

This is our cheap solution so far to this problem.  Normally users need to restart to get the new Jupyter env release only.  They will not restart just for fun so I think that's why we did not need any better solutions.

File: Dockerfile (hunk 3 above)
Comment (ChaamC) [reply Dockerfile:60]: Yes, that is correct. The last used version of the base image is 0.2.2 and it was built on Sep 14, 2021 at 3:18 pm. I took the last commit found on the `docker-stacks` repo from that date.

**Review by matprov — APPROVED** [review]

**Review by fmigneault — APPROVED** [review]

**Pull request merged**